### ISSUE TITLE
Webpack

### ISSUE THREAD

**Issue by Apidcloud** [dan-kez/lerna-webpack-example#4]
Hello! I am trying to figure out the best way to configure webpack and typescript with lerna.

Could you include the scripts and configs you use for webpack too?

Thank you!

**Comment by dan-kez** [comment]
Hi @Apidcloud 

I’m a little confused by the question- the repository includes a webpack config, tsconfig, and Babel.config.js which all work in concert.

What specific issue are you running into?

**Comment by Apidcloud** [comment]
But I think there is no webpack config in the repo. Is it perhaps in another branch?

**Comment by dan-kez** [comment]
It’s here : https://github.com/dan-kez/lerna-webpack-example/blob/master/packages/router/webpack.config.js

**Comment by Apidcloud** [comment]
Ah, thank you! Dunno how I missed that.

**Comment by dan-kez** [comment]
No worries- let me know if you have any questions!

**Comment by Apidcloud** [comment]
I still wasn't able to configure the structure. I'm only using typescript for type checking (babel for compiling), but for some reason when it type checks a package (**core**) that depends on other (**commons**), it says that the files (**from** **commons**) are not in the project file list. Projects must list all files or use an 'include' pattern. It's weird.

But the type checking for **commons** package works as expected. If I try to use the _references_ option within **core ts.config** with the **path of commons folder**, then it starts saying that the module could not be found. Also weird.

Could you please take a look? I have been at it for hours, but have no clue.
I basically want to do type checking using `yarn check-types` and emit the typings with `yarn build:types` to a `dist` folder.

https://github.com/scarlettgamestudio/scarlett-framework-2/tree/chore/typescript

**Comment by dan-kez** [comment]
@Apidcloud I recommend having only one webpack config across all packages (e.g. what you see here) unless you need to build for a completely different use case. As an example, at Klaviyo we have 2 webpack configs: one for the main product application, and one for JS assets that are served on our customer's websites.

It depends on what you're looking to do.

**Comment by Apidcloud** [comment]
Thanks, @dan-kez 

I ended up doing a global automated config, while still allowing custom configs per package when needed. I'm bundling as UMD so it works both on the browser and node, but I'm not sure how to put all the packages under the same global name while in the browser. I just noticed this yesterday, but as it is, it replaces the global variable entirely—it doesn't reuse it, so the last script tag is the one that I am able to use. Or maybe I should think of a better naming alternative for the browser...
